frontend/src/container/MetricsApplication/Tabs/External/Table/index.tsx

```diff
@@ -0,0 +1,187 @@
+import { ResizeTable } from 'components/ResizeTable';
+import { useNotifications } from 'hooks/useNotifications';
+import React, { useMemo } from 'react';
+import Highlighter from 'react-highlight-words';
+import { useTranslation } from 'react-i18next';
+import { useQuery } from 'react-query';
+import { useSelector } from 'react-redux';
+import {
+	GetMetricQueryRange,
+	GetQueryResultsProps,
+} from 'store/actions/dashboard/getQueryResults';
+import { AppState } from 'store/reducers';
+import { ErrorResponse, SuccessResponse } from 'types/api';
+import { MetricRangePayloadProps } from 'types/api/metrics/getQueryRange';
+import { GlobalReducer } from 'types/reducer/globalTime';
+
+import { makeTableRows, TableRow } from './calculations';
+import {
+	makeDurationQuery,
+	makeErrPercentQuery,
+	makeReqRateQuery,
+} from './queries';
+import { ReduceTo } from './reduceTo';
+import Toolbar from './Toolbar';
+
+interface TableProps {
+	widgetId: string;
+	serviceName: string;
+}
+
+const noDataPlaceholder = '-';
+
+function Table(props: TableProps): JSX.Element {
+	const { widgetId, serviceName } = props;
+	const { selectedTime } = useSelector<AppState, GlobalReducer>(
+		(state) => state.globalTime,
+	);
+	const [addressFilter, setAddressFilter] = React.useState<string>('');
+	const [reduceTo, setReduceTo] = React.useState<ReduceTo>('latest');
+
+	const { t } = useTranslation(['common']);
+	const { notifications } = useNotifications();
+	const errorMessage = t('something_went_wrong');
+
+	// BEGIN Request rate
+	const reqRateQuery: GetQueryResultsProps = makeReqRateQuery({
+		widgetId: `${widgetId}-req-rate`,
+		selectedTime,
+		serviceName,
+	});
+
+	const reqRateResponse = useQuery<
+		SuccessResponse<MetricRangePayloadProps> | ErrorResponse
+	>(`${widgetId}-req-rate`, () => GetMetricQueryRange(reqRateQuery));
+
+	if (reqRateResponse.isError) {
+		notifications.error({
```

Comment:
IMO if the component re-render with the same condition there would be a chain of notifications

Reply:
Fixed